cmd/send/send.go

```
@@ -51,5 +52,16 @@ func parseArgumentsAndCallSend(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("getting stage artifacts root failed, original error: '%w'", err)
 	}
 
-	return sendImplementation.PrepareAndSendSubmissionInformationPackage(common_flags.KafkaEndpoints, transferTopicName, stageArtifactsRoot)
+	err = sendImplementation.PrepareAndSendSubmissionInformationPackage(common_flags.KafkaEndpoints, transferTopicName, stageArtifactsRoot)
+	if err != nil {
+		err = fmt.Errorf("transfer error, cause: `%w`", err)
+		fmt.Printf("Sending error message to Teams\n")
+		teamsErrorMessage := teams.CreateGeneralFailureMessage(err)
+		if err := teams.SendMessage(teamsErrorMessage, common_flags.TeamsWebhookNotificationUrl); err != nil {
+			err = fmt.Errorf("failed to send error message to Teams, cause: `%w`", err)
+			fmt.Printf("%s\n", err)
+		}
+		return err
```

This error message should just be logged and not returned, because we would like to return the original error message and not the error that the teams message failed to be sent.

---

If we are failing to send a message to teams, we would also know that it failed, otherwise how would we know that it failed? IMO we are simply wrapping the original error with the teams error, so we will get a nested error message that contains all of the information.

Say you have the scenario where an original error moves through a system and triggers other errors along the way:

```go
func someFunc01() error {
    print("someFunc01")
    return fmt.Errorf("someFunc01 error")
}

func someFunc02() error {
    err := someFunc01()
    if err != nil {
        return fmt.Errorf("someFunc02 error: %w", err) // not appending the error here would be a mistake IMO
    }
    return nil
}

func main() {
    err := someFunc02()
    if err != nil {
        fmt.Println(err)
    }
}
```

It is crucial for a developer to know what went wrong, and silently dropping an error message can lead to confusion. 

Or am I misunderstanding something

---

I misread the code which is actually doing what I proposed.